integtest/src/main/java/org/corfudb/universe/cluster/Cluster.java

```diff
@@ -0,0 +1,95 @@
+package org.corfudb.universe.cluster;
+
+import com.google.common.collect.ImmutableMap;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import org.corfudb.universe.node.CorfuClient;
+import org.corfudb.universe.node.CorfuServer;
+import org.corfudb.universe.node.Node;
+import org.corfudb.universe.service.Service;
+import org.corfudb.universe.service.Service.ServiceParams;
+
+import java.time.Duration;
+import java.util.UUID;
+
+import static lombok.Builder.Default;
+import static lombok.EqualsAndHashCode.Exclude;
+import static org.corfudb.universe.node.Node.NodeParams;
+
+/**
+ * A Cluster represents a common notion of a cluster of nodes. The architecture of a cluster is composed of collections
+ * of {@link Service}s and {@link Node}s.
+ * Each instance of service in the cluster is composed of a collection of {@link Node}s which are a subset of the
+ * entire nodes included in the cluster.
+ * Cluster configuration is provided by an instance of {@link ClusterParams}.
+ * <p>
+ * The followings are the main functionalities provided by this class:
+ * DEPLOY: create a cluster according to cluster parameters.
+ * SHUTDOWN: shutdown a cluster by stopping the nodes, and shutdown the network.
+ */
+public interface Cluster {
+
+    /**
+     * Create a cluster according to the desired state mentioned by {@link ClusterParams}
+     *
+     * @return an immutable instance of applied change in cluster
+     * @throws ClusterException
+     */
+    Cluster deploy() throws ClusterException;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.corfudb.universe.cluster.ClusterException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)